### PR TITLE
User templates on index.html

### PR DESCRIPTION
**Pull request by cmbrose**
This will hopefully help solve the auth refreshing issue by adding a template to `index.html` to store the backend's current user's email.  The template is populated by the backend.

### Testing

Three main scenarios should be covered:

1. standard full page refresh
2. lighthouse restarts, user refreshes (should redirect to /login)
3. lighthouse restarts, user attempts API call (should redirect to /login)

### CODE REVIEW

**Comment by cmbrose** [comment]
https://github.com/lighthouse/lighthouse/pull/51

File: app/index.html [hunk 1]
```diff
@@ -25,6 +25,15 @@
         padding-right: 15px;
       }
     </style>
+
+    <!-- Current user's email from Lighthouse -->
+    <script>
+      {{ if .LoggedIn }}
+        window.user = {"{{.Email}}"};
```
Comment (ngmiller) [inline app/index.html]: This should probably be formatted as `{email: "{{.Email}}"};`

**Comment by ngmiller** [comment]
@theMagicalKarp @zach-taylor

**Comment by ngmiller** [comment]
@theMagicalKarp @zach-taylor Note that to make this work without generating a bunch of alerts with repeated information, I had to modify the $http error handling semantics a bit. You'll notice that I've removed all the error callbacks from each localized $http call, which are now handling in a single location (the httpInterceptor in app.js)

I'm hoping this won't be a major problem down the road since the majority of our error handlers followed the same format anyway:
```
alertService.create({
    message: response.data,
    type: ...
});
```

If we can work with the backend to ensure each API error response is in a consistent format, we can build different responses based on the error type ("control", "docker", etc) coming back

**Comment by zach-taylor** [comment]
Works for me. +1

**Pull request merged**